File: src/main/java/com/polygloat/dtos/request/CreateRepositoryDTO.java
package com.polygloat.dtos.request;

import com.polygloat.dtos.request.validators.annotations.RepositoryRequest;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Set;

@RepositoryRequest
@NoArgsConstructor
public class CreateRepositoryDTO extends AbstractRepositoryDTO {
    @NotEmpty
    @Getter
    @Setter
    Set<LanguageDTO> languages;

    public CreateRepositoryDTO(@NotNull String name, @NotEmpty Set<LanguageDTO> languages) {
        this.name = name;
        this.languages = languages;
    }
}

File: src/main/java/io/tolgee/dtos/request/CreateRepositoryDTO.kt
package io.tolgee.dtos.request

import io.swagger.v3.oas.annotations.media.Schema
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

data class CreateRepositoryDTO(
        @field:NotNull @field:Size(min = 3, max = 50)
        var name: String? = null,

        @field:NotEmpty
        var languages: Set<LanguageDTO>? = null,

        @field:Size(min = 3, max = 60)
        @field:Pattern(regexp = "^[a-z0-9-]*[a-z]+[a-z0-9-]*$", message = "invalid_pattern")
        var addressPart: String? = null,

        @Schema(description = "If not provided, project will be created as users")
        var organizationId: Long? = null
)
